baby-liminal-extension/src/executor.rs

```diff
@@ -1,15 +1,14 @@
-use frame_support::dispatch::Weight;
-use pallet_baby_liminal::{Error, Pallet as BabyLiminal, ProvingSystem, VerificationKeyIdentifier};
-use sp_std::vec::Vec;
+use obce::substrate::{
+    frame_support::weights::Weight,
+    frame_system::Config as SysConfig,
+    pallet_contracts::chain_extension::{BufInBufOutState, Environment, Ext},
+};
+use pallet_baby_liminal::{Config as BabyLiminalConfig, Error, Pallet as BabyLiminal};
 
-use crate::Runtime;
+use crate::{ProvingSystem, Vec, VerificationKeyIdentifier};
 
-/// Abstraction around `Runtime`. Makes testing easier.
-///
-/// Gathers all the methods that are used by `BabyLiminalChainExtension`.
-///
-/// Each method is already documented in `pallet_baby_liminal`.
-pub(super) trait Executor: Sized {
+/// Generalized pallet executor, that can be mocked for testing purposes.
+pub trait Executor<T>: Sized {
```

Comment:
but this particular trait doesn't have to be, does it?

Reply:
Why not? This trait generalizes pallet-specific code, so it has two impls - one for the production, and one for testing. The production one is `impl<'a, 'b, E, T> Executor<T> for Environment<'a, 'b, E, BufInBufOutState>`, and we cannot remove generic `T` from here, because we don't know the concrete runtime type to use inside trait methods.

If this trait was used for testing purposes only, the yes, we could remove that generic, but since there are two locations where it's used, with one location without access to any concrete runtime type whatsoever (since `aleph-runtime` is a `dev-dependency`, and since chain extensions are generic over runtime anyway).